test/powershell/Language/LanguageTestSupport.psm1

```diff
@@ -24,18 +24,57 @@ function Get-RuntimeError
     [CmdletBinding()]
     param(
         [Parameter(ValueFromPipeline=$True,Mandatory=$True)]
-        [string]$src
+        [string]$src,
+        [Parameter()]
+        [int]$Timeout = 0
     )
-
+ 
     $errors = $null
     try 
     {
-        [scriptblock]::Create($src).Invoke() > $null
+        # some tests cannot be run in a isolated runspace
+        # easily because they require more than just a single
+        # execution
+        if ( $Timeout -eq 0 )
+        {
+            [scriptblock]::Create($src).Invoke() > $null
+        }
+        else
+        {
+            $ps = [powershell]::Create()
+            $ps.AddScript($src) > $null
+            $ar = $ps.BeginInvoke()
+            # give it 250 milliseconds to complete
+            start-sleep -mill 250

```

Comment:
yes, it will slow down some tests, but I reckoned that it was better to slow down the test than to invalidate an entire run - that's what I'm trying to avoid. When running a full test pass I saw nearly 100% test runs killed for lack of activity. Each one of those runs were hung in the language tests.

---
In reply to: [94967709](https://github.com/PowerShell/PowerShell/pull/2958#discussion_r94967709) [](ancestors = 94967709)